app/assets/javascripts/discourse/controllers/preferences/account.js.es6

```diff
@@ -135,6 +170,26 @@ export default Ember.Controller.extend(
 
       showTwoFactorModal() {
         showModal("second-factor-intro");
+      },
+
+      revokeAccount(account) {
+        const model = this.get("model");
+        this.set("revoking", true);
+        model
+          .revokeAssociatedAccount(account.name)
+          .then(result => {
+            if (result.success) {
+              model.get("associated_accounts").removeObject(account);
+            } else {
+              bootbox.alert(result.message);
+            }
+            this.set("revoking", false);
```

Comment:
Not sure about all the logic here, but should this be in a finally block instead?